Build failure ( zypp backend )

Compiled on `Debian unstable`:

```
$ ninja
[230/231] Compiling C++ object backends/zypp/libpk_backend_zypp.so.p/pk-backend-zypp.cpp.o
../backends/zypp/pk-backend-zypp.cpp:637:1: warning: no previous declaration for ‘gboolean zypp_logging()’ [-Wmissing-declarations]
  637 | zypp_logging ()
      | ^~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:683:1: warning: no previous declaration for ‘gboolean zypp_is_development_repo(zypp::RepoInfo)’ [-Wmissing-declarations]
  683 | zypp_is_development_repo (RepoInfo repo)
      | ^~~~~~~~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:692:1: warning: no previous declaration for ‘gboolean zypp_is_valid_repo(PkBackendJob*, zypp::RepoInfo)’ [-Wmissing-declarations]
  692 | zypp_is_valid_repo (PkBackendJob *job, RepoInfo repo)
      | ^~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:713:1: warning: no previous declaration for ‘zypp::ResPool zypp_build_pool(zypp::ZYpp::Ptr, gboolean)’ [-Wmissing-declarations]
  713 | zypp_build_pool (ZYpp::Ptr zypp, gboolean include_local)
      | ^~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:774:1: warning: no previous declaration for ‘zypp::target::rpm::RpmHeader::constPtr zypp_get_rpmHeader(const std::string&, zypp::Edition)’ [-Wmissing-declarations]
  774 | zypp_get_rpmHeader (const string &name, Edition edition)
      | ^~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:790:1: warning: no previous declaration for ‘PkGroupEnum get_enum_group(const std::string&)’ [-Wmissing-declarations]
  790 | get_enum_group (const string &group_)
      | ^~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:845:1: warning: no previous declaration for ‘void zypp_get_packages_by_name(const gchar*, zypp::ResKind, std::vector<zypp::sat::Solvable>&, gboolean)’ [-Wmissing-declarations]
  845 | zypp_get_packages_by_name (const gchar *package_name,
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:867:1: warning: no previous declaration for ‘void zypp_get_packages_by_file(zypp::ZYpp::Ptr, const gchar*, std::vector<zypp::sat::Solvable>&)’ [-Wmissing-declarations]
  867 | zypp_get_packages_by_file (ZYpp::Ptr zypp,
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:899:1: warning: no previous declaration for ‘bool zypp_package_is_local(const gchar*)’ [-Wmissing-declarations]
  899 | zypp_package_is_local (const gchar *package_id)
      | ^~~~~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:920:1: warning: no previous declaration for ‘zypp::sat::Solvable zypp_get_package_by_id(const gchar*)’ [-Wmissing-declarations]
  920 | zypp_get_package_by_id (const gchar *package_id)
      | ^~~~~~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:985:1: warning: no previous declaration for ‘zypp::RepoInfo zypp_get_Repository(PkBackendJob*, const gchar*)’ [-Wmissing-declarations]
  985 | zypp_get_Repository (PkBackendJob *job, const gchar *alias)
      | ^~~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:1175:1: warning: no previous declaration for ‘void zypp_emit_filtered_packages_in_list(PkBackendJob*, PkBitfield, const std::vector<zypp::sat::Solvable>&)’ [-Wmissing-declarations]
 1175 | zypp_emit_filtered_packages_in_list (PkBackendJob *job, PkBitfield filters, const vector<sat::Solvable> &v)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp: In function ‘void zypp_backend_finished_error(PkBackendJob*, PkErrorEnum, const char*, ...)’:
../backends/zypp/pk-backend-zypp.cpp:1803:35: warning: function ‘void zypp_backend_finished_error(PkBackendJob*, PkErrorEnum, const char*, ...)’ might be a candidate for ‘gnu_printf’ format attribute [-Wsuggest-attribute=format]
 1803 |         buffer = g_strdup_vprintf (format, args);
      |                  ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp: In function ‘void pk_backend_upgrade_system_thread(PkBackendJob*, GVariant*, gpointer)’:
../backends/zypp/pk-backend-zypp.cpp:3576:23: warning: unused variable ‘restart’ [-Wunused-variable]
 3576 |         PkRestartEnum restart = PK_RESTART_ENUM_NONE;
      |                       ^~~~~~~
../backends/zypp/pk-backend-zypp.cpp: In function ‘void backend_what_provides_thread(PkBackendJob*, GVariant*, gpointer)’:
../backends/zypp/pk-backend-zypp.cpp:3846:62: warning: type qualifiers ignored on cast result type [-Wignored-qualifiers]
 3846 |                                                              (const gpointer) make<ResObject>(*it)->summary().c_str (),
      |                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[231/231] Linking target backends/zypp/libpk_backend_zypp.so
```

Build now fails with 45d31408e.

Below log from zypp CI job - https://github.com/sidt4/PackageKit/actions/runs/10537898788/job/29199803510

```
FAILED: backends/zypp/libpk_backend_zypp.so.p/pk-backend-zypp.cpp.o 
c++ -Ibackends/zypp/libpk_backend_zypp.so.p -Ibackends/zypp -I../backends/zypp -Isrc -I../src -Ilib/packagekit-glib2 -I../lib/packagekit-glib2 -Ilib -I../lib -I. -I.. -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/usr/include/x86_64-linux-gnu -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -Wall -Winvalid-pch -Wextra -O0 -g -DPK_COMPILATION -Wall -Wcast-align -Wno-uninitialized -Werror=missing-declarations -Wredundant-decls -Wpointer-arith -Wcast-align -Wwrite-strings -Winit-self -Wreturn-type -Wformat-nonliteral -Wformat-security -Wmissing-include-dirs -Wmissing-format-attribute -Wclobbered -Wempty-body -Wignored-qualifiers -Wsign-compare -Wtype-limits -Wuninitialized -Wno-unused-parameter -fPIC -pthread -D_FILE_OFFSET_BITS=64 '-DG_LOG_DOMAIN="PackageKit-Zypp"' -Wall -Woverloaded-virtual -Wnon-virtual-dtor -std=c++1z -MD -MQ backends/zypp/libpk_backend_zypp.so.p/pk-backend-zypp.cpp.o -MF backends/zypp/libpk_backend_zypp.so.p/pk-backend-zypp.cpp.o.d -o backends/zypp/libpk_backend_zypp.so.p/pk-backend-zypp.cpp.o -c ../backends/zypp/pk-backend-zypp.cpp
../backends/zypp/pk-backend-zypp.cpp:637:1: error: no previous declaration for 'gboolean zypp_logging()' [-Werror=missing-declarations]
  637 | zypp_logging ()
      | ^~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:683:1: error: no previous declaration for 'gboolean zypp_is_development_repo(zypp::RepoInfo)' [-Werror=missing-declarations]
  683 | zypp_is_development_repo (RepoInfo repo)
      | ^~~~~~~~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:692:1: error: no previous declaration for 'gboolean zypp_is_valid_repo(PkBackendJob*, zypp::RepoInfo)' [-Werror=missing-declarations]
  692 | zypp_is_valid_repo (PkBackendJob *job, RepoInfo repo)
      | ^~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:713:1: error: no previous declaration for 'zypp::ResPool zypp_build_pool(zypp::ZYpp::Ptr, gboolean)' [-Werror=missing-declarations]
  713 | zypp_build_pool (ZYpp::Ptr zypp, gboolean include_local)
      | ^~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:774:1: error: no previous declaration for 'zypp::target::rpm::RpmHeader::constPtr zypp_get_rpmHeader(const std::string&, zypp::Edition)' [-Werror=missing-declarations]
  774 | zypp_get_rpmHeader (const string &name, Edition edition)
      | ^~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:790:1: error: no previous declaration for 'PkGroupEnum get_enum_group(const std::string&)' [-Werror=missing-declarations]
  790 | get_enum_group (const string &group_)
      | ^~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:845:1: error: no previous declaration for 'void zypp_get_packages_by_name(const gchar*, zypp::ResKind, std::vector<zypp::sat::Solvable>&, gboolean)' [-Werror=missing-declarations]
  845 | zypp_get_packages_by_name (const gchar *package_name,
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:867:1: error: no previous declaration for 'void zypp_get_packages_by_file(zypp::ZYpp::Ptr, const gchar*, std::vector<zypp::sat::Solvable>&)' [-Werror=missing-declarations]
  867 | zypp_get_packages_by_file (ZYpp::Ptr zypp,
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:899:1: error: no previous declaration for 'bool zypp_package_is_local(const gchar*)' [-Werror=missing-declarations]
  899 | zypp_package_is_local (const gchar *package_id)
      | ^~~~~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:920:1: error: no previous declaration for 'zypp::sat::Solvable zypp_get_package_by_id(const gchar*)' [-Werror=missing-declarations]
  920 | zypp_get_package_by_id (const gchar *package_id)
      | ^~~~~~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:985:1: error: no previous declaration for 'zypp::RepoInfo zypp_get_Repository(PkBackendJob*, const gchar*)' [-Werror=missing-declarations]
  985 | zypp_get_Repository (PkBackendJob *job, const gchar *alias)
      | ^~~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp:1175:1: error: no previous declaration for 'void zypp_emit_filtered_packages_in_list(PkBackendJob*, PkBitfield, const std::vector<zypp::sat::Solvable>&)' [-Werror=missing-declarations]
 1175 | zypp_emit_filtered_packages_in_list (PkBackendJob *job, PkBitfield filters, const vector<sat::Solvable> &v)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp: In function 'void zypp_backend_finished_error(PkBackendJob*, PkErrorEnum, const char*, ...)':
../backends/zypp/pk-backend-zypp.cpp:1803:35: warning: function 'void zypp_backend_finished_error(PkBackendJob*, PkErrorEnum, const char*, ...)' might be a candidate for 'gnu_printf' format attribute [-Wsuggest-attribute=format]
 1803 |         buffer = g_strdup_vprintf (format, args);
      |                  ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
../backends/zypp/pk-backend-zypp.cpp: In function 'void pk_backend_upgrade_system_thread(PkBackendJob*, GVariant*, gpointer)':
../backends/zypp/pk-backend-zypp.cpp:3576:23: warning: unused variable 'restart' [-Wunused-variable]
 3576 |         PkRestartEnum restart = PK_RESTART_ENUM_NONE;
      |                       ^~~~~~~
../backends/zypp/pk-backend-zypp.cpp: In function 'void backend_what_provides_thread(PkBackendJob*, GVariant*, gpointer)':
../backends/zypp/pk-backend-zypp.cpp:3847:62: warning: type qualifiers ignored on cast result type [-Wignored-qualifiers]
 3847 |                                                              (const gpointer) make<ResObject>(*it)->summary().c_str (),
      |                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: some warnings being treated as errors
ninja: build stopped: subcommand failed.
Error: Process completed with exit code 1.
```

--- a/backends/zypp/pk-backend-zypp.cpp
+++ b/backends/zypp/pk-backend-zypp.cpp
@@ -3569,7 +3569,6 @@ pk_backend_upgrade_system_thread (PkBackendJob *job,
 	}
 
 	ResPool pool = zypp_build_pool (zypp, TRUE);
-	PkRestartEnum restart = PK_RESTART_ENUM_NONE;
 	PoolStatusSaver saver;
 
 	if (is_tumbleweed ()) {

--- a/backends/zypp/pk-backend-zypp.cpp
+++ b/backends/zypp/pk-backend-zypp.cpp
@@ -633,7 +633,7 @@ ZyppJob::get_zypp()
 /**
   * Enable and rotate zypp logging
   */
-gboolean
+static gboolean
 zypp_logging ()
 {
 	gchar *file = g_strdup ("/var/log/pk_backend_zypp");
@@ -679,7 +679,7 @@ namespace {
 	}
 }
 
-gboolean
+static gboolean
 zypp_is_development_repo (RepoInfo repo)
 {
 	return ( name_ends_or_contains( repo.alias(), "-debuginfo" )
@@ -688,7 +688,7 @@ zypp_is_development_repo (RepoInfo repo)
 	      || name_ends_or_contains( repo.alias(), "-development" ) );
 }
 
-gboolean
+static gboolean
 zypp_is_valid_repo (PkBackendJob *job, RepoInfo repo)
 {
 
@@ -709,7 +709,7 @@ zypp_is_valid_repo (PkBackendJob *job, RepoInfo repo)
  * Build and return a ResPool that contains all local resolvables
  * and ones found in the enabled repositories.
  */
-ResPool
+static ResPool
 zypp_build_pool (ZYpp::Ptr zypp, gboolean include_local)
 {
 	static gboolean repos_loaded = FALSE;
@@ -770,7 +770,7 @@ zypp_build_pool (ZYpp::Ptr zypp, gboolean include_local)
 /**
   * Return the rpmHeader of a package
   */
-target::rpm::RpmHeader::constPtr
+static target::rpm::RpmHeader::constPtr
 zypp_get_rpmHeader (const string &name, Edition edition)
 {
 	target::rpm::librpmDb::db_const_iterator it;
@@ -786,7 +786,7 @@ zypp_get_rpmHeader (const string &name, Edition edition)
 /**
   * Return the PkEnumGroup of the given PoolItem.
   */
-PkGroupEnum
+static PkGroupEnum
 get_enum_group (const string &group_)
 {
 	string group(str::toLower(group_));
@@ -841,7 +841,7 @@ get_enum_group (const string &group_)
 /**
  * Returns a list of packages that match the specified package_name.
  */
-void
+static void
 zypp_get_packages_by_name (const gchar *package_name,
 			   const ResKind kind,
 			   vector<sat::Solvable> &result,
@@ -863,7 +863,7 @@ zypp_get_packages_by_name (const gchar *package_name,
 /**
  * Returns a list of packages that owns the specified file.
  */
-void
+static void
 zypp_get_packages_by_file (ZYpp::Ptr zypp,
 			   const gchar *search_file,
 			   vector<sat::Solvable> &ret)
@@ -895,7 +895,7 @@ zypp_get_packages_by_file (ZYpp::Ptr zypp,
 /**
  * Return the package is from a local file or not.
  */
-bool
+static bool
 zypp_package_is_local (const gchar *package_id)
 {
 	MIL << package_id << endl;
@@ -916,7 +916,7 @@ zypp_package_is_local (const gchar *package_id)
  * Returns the Resolvable for the specified package_id.
  * e.g. gnome-packagekit;3.6.1-132.1;x86_64;G:F
 */
-sat::Solvable
+static sat::Solvable
 zypp_get_package_by_id (const gchar *package_id)
 {
 	MIL << package_id << endl;
@@ -981,7 +981,7 @@ zypp_get_package_by_id (const gchar *package_id)
 	return package;
 }
 
-RepoInfo
+static RepoInfo
 zypp_get_Repository (PkBackendJob *job, const gchar *alias)
 {
 	RepoInfo info;
@@ -1171,7 +1171,7 @@ zypp_backend_package (PkBackendJob *job, PkInfoEnum info,
  * we don't notify the client that the package is also available, since
  * PK doesn't handle re-installs (by some quirk).
  */
-void
+static void
 zypp_emit_filtered_packages_in_list (PkBackendJob *job, PkBitfield filters, const vector<sat::Solvable> &v)
 {
 	typedef vector<sat::Solvable>::const_iterator sat_it_t;

--- a/backends/zypp/pk-backend-zypp.cpp
+++ b/backends/zypp/pk-backend-zypp.cpp
@@ -3839,7 +3839,7 @@ backend_what_provides_thread (PkBackendJob *job, GVariant *params, gpointer user
 			for (sat::WhatProvides::const_iterator it = prov.begin (); it != prov.end (); ++it) {
 				if (it->isSystem ())
 					g_hash_table_insert (installed_hash,
-							     (const gpointer) make<ResObject>(*it)->summary().c_str (),
+							     (gpointer) make<ResObject>(*it)->summary().c_str (),
 							     GUINT_TO_POINTER (1));
 			}
 

--- a/backends/zypp/pk-backend-zypp.cpp
+++ b/backends/zypp/pk-backend-zypp.cpp
@@ -863,6 +863,7 @@ zypp_get_packages_by_name (const gchar *package_name,
 /**
  * Returns a list of packages that owns the specified file.
  */
+G_GNUC_UNUSED
 static void
 zypp_get_packages_by_file (ZYpp::Ptr zypp,
 			   const gchar *search_file,

--- a/backends/zypp/pk-backend-zypp.cpp
+++ b/backends/zypp/pk-backend-zypp.cpp
@@ -1792,6 +1792,7 @@ zypp_refresh_cache (PkBackendJob *job, ZYpp::Ptr zypp, gboolean force)
 /**
   * helper to simplify returning errors
   */
+G_GNUC_PRINTF (3, 4)
 static void
 zypp_backend_finished_error (PkBackendJob  *job, PkErrorEnum err_code,
 			     const char *format, ...)
@@ -1931,7 +1932,7 @@ backend_required_by_thread (PkBackendJob *job, GVariant *params, gpointer user_d
 			}
 			zypp_backend_finished_error (
 				job, PK_ERROR_ENUM_DEP_RESOLUTION_FAILED,
-				problem.c_str());
+				"%s", problem.c_str());
 			return;
 		}
 
@@ -2128,11 +2129,11 @@ backend_depends_on_thread (PkBackendJob *job, GVariant *params, gpointer user_da
 		pk_backend_job_set_percentage (job, 100);
 	} catch (const repo::RepoNotFoundException &ex) {
 		zypp_backend_finished_error (
-			job, PK_ERROR_ENUM_REPO_NOT_FOUND, ex.asUserString().c_str());
+			job, PK_ERROR_ENUM_REPO_NOT_FOUND, "%s", ex.asUserString().c_str());
 		return;
 	} catch (const Exception &ex) {
 		zypp_backend_finished_error (
-			job, PK_ERROR_ENUM_INTERNAL_ERROR, ex.asUserString().c_str());
+			job, PK_ERROR_ENUM_INTERNAL_ERROR, "%s", ex.asUserString().c_str());
 		return;
 	}
 }
@@ -2212,7 +2213,7 @@ backend_get_details_thread (PkBackendJob *job, GVariant *params, gpointer user_d
 				(gulong)size);
 		} catch (const Exception &ex) {
 			zypp_backend_finished_error (
-				job, PK_ERROR_ENUM_INTERNAL_ERROR, ex.asUserString ().c_str ());
+				job, PK_ERROR_ENUM_INTERNAL_ERROR, "%s", ex.asUserString ().c_str ());
 			return;
 		}
 	}
@@ -2594,7 +2595,7 @@ backend_install_files_thread (PkBackendJob *job, GVariant *params, gpointer user
 
 	} catch (const Exception &ex) {
 		zypp_backend_finished_error (
-			job, PK_ERROR_ENUM_INTERNAL_ERROR, ex.asUserString ().c_str ());
+			job, PK_ERROR_ENUM_INTERNAL_ERROR, "%s", ex.asUserString ().c_str ());
 		return;
 	}
 
@@ -2866,7 +2867,7 @@ backend_install_packages_thread (PkBackendJob *job, GVariant *params, gpointer u
 
 	} catch (const Exception &ex) {
 		zypp_backend_finished_error (
-			job, PK_ERROR_ENUM_INTERNAL_ERROR, ex.asUserString().c_str());
+			job, PK_ERROR_ENUM_INTERNAL_ERROR, "%s", ex.asUserString().c_str());
 		return;
 	}
 }
@@ -2982,11 +2983,11 @@ backend_remove_packages_thread (PkBackendJob *job, GVariant *params, gpointer us
 
 	} catch (const repo::RepoNotFoundException &ex) {
 		zypp_backend_finished_error (
-			job, PK_ERROR_ENUM_REPO_NOT_FOUND, ex.asUserString().c_str());
+			job, PK_ERROR_ENUM_REPO_NOT_FOUND, "%s", ex.asUserString().c_str());
 		return;
 	} catch (const Exception &ex) {
 		zypp_backend_finished_error (
-			job, PK_ERROR_ENUM_INTERNAL_ERROR, ex.asUserString().c_str());
+			job, PK_ERROR_ENUM_INTERNAL_ERROR, "%s", ex.asUserString().c_str());
 		return;
 	}
 }
@@ -3276,11 +3277,11 @@ pk_backend_get_repo_list (PkBackend *backend, PkBackendJob *job, PkBitfield filt
 		repos = list<RepoInfo>(manager.repoBegin(),manager.repoEnd());
 	} catch (const repo::RepoNotFoundException &ex) {
 		zypp_backend_finished_error (
-			job, PK_ERROR_ENUM_REPO_NOT_FOUND, ex.asUserString().c_str());
+			job, PK_ERROR_ENUM_REPO_NOT_FOUND, "%s", ex.asUserString().c_str());
 		return;
 	} catch (const Exception &ex) {
 		zypp_backend_finished_error (
-			job, PK_ERROR_ENUM_INTERNAL_ERROR, ex.asUserString().c_str());
+			job, PK_ERROR_ENUM_INTERNAL_ERROR, "%s", ex.asUserString().c_str());
 		return;
 	}
 
@@ -3331,11 +3332,11 @@ pk_backend_repo_enable (PkBackend *backend, PkBackendJob *job, const gchar *rid,
 
 	} catch (const repo::RepoNotFoundException &ex) {
 		zypp_backend_finished_error (
-			job, PK_ERROR_ENUM_REPO_NOT_FOUND, ex.asUserString().c_str());
+			job, PK_ERROR_ENUM_REPO_NOT_FOUND, "%s", ex.asUserString().c_str());
 		return;
 	} catch (const Exception &ex) {
 		zypp_backend_finished_error (
-			job, PK_ERROR_ENUM_INTERNAL_ERROR, ex.asUserString().c_str());
+			job, PK_ERROR_ENUM_INTERNAL_ERROR, "%s", ex.asUserString().c_str());
 		return;
 	}
 
@@ -3953,7 +3954,7 @@ backend_download_packages_thread (PkBackendJob *job, GVariant *params, gpointer 
 		}
 	} catch (const Exception &ex) {
 		zypp_backend_finished_error (
-			job, PK_ERROR_ENUM_PACKAGE_DOWNLOAD_FAILED, ex.asUserString().c_str());
+			job, PK_ERROR_ENUM_PACKAGE_DOWNLOAD_FAILED, "%s", ex.asUserString().c_str());
 		return;
 	}
 }

